src/executables/csql.c

```diff
@@ -3330,3 +3358,131 @@ get_host_ip (unsigned char *ip_addr)
 
   return 0;
 }
+
+static int
+csql_connect (char *argument, CSQL_ARGUMENT * csql_arg)
+{
+  /*dbname can be stored DB name+ @ + remote hostname + \0 */
+  char buf[DB_MAX_USER_LENGTH + MAXLOGNAME + CUB_MAXHOSTNAMELEN + 1];
+  const char *delim = " \n";
+  char *save_ptr_strtok = NULL;
+  const char *db_name_ptr = NULL;
+  char *user_name_ptr = NULL;
+  char *host_name_ptr = NULL;
+  char *p = NULL;
+  const char *err_msg;
+  int error_code;
+  bool mem_alloc = false;
+  CSQL_ARGUMENT csql_new_arg;
+
+  if (argument == NULL)
+    {
+      err_msg = (*csql_get_message) (CSQL_MSG_TOO_FEW_ARGS);
+      fprintf (stderr, err_msg);
+      return DO_CMD_SUCCESS;
+    }
+
+  /*verify @hostname */
+  if ((host_name_ptr = strchr (argument, '@')) != NULL && (*(host_name_ptr + 1) == '\0'))
+    {
+      csql_Error_code = CSQL_ERR_SQL_ERROR;
+      return DO_CMD_FAILURE;
+    }
+
+  strcpy (buf, argument);
+
+  if ((user_name_ptr = strtok_r (buf, delim, &save_ptr_strtok)) == NULL)
+    {
+      csql_Error_code = CSQL_ERR_SQL_ERROR;
+      return DO_CMD_FAILURE;
+    }
+
+  /*find db name following the user name */
+  db_name_ptr = strtok_r (NULL, delim, &save_ptr_strtok);
+  if (db_name_ptr == NULL)
+    {
+      db_name_ptr = csql_arg->db_name;
+    }
+
+  memcpy (&csql_new_arg, csql_arg, sizeof (CSQL_ARGUMENT));
+
+#if defined(CS_MODE)
+  memset (boot_Host_connected, 0, sizeof (boot_Host_connected));
```

Comment:
Is this code necessary?

Reply:
Yes, If not, It cannot works to connect other hosts.